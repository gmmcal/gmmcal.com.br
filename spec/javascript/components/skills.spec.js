import { mount, shallowMount } from '@vue/test-utils'
import Skill from '@/components/skill'
import Skills from '@/components/skills'
import create from '../settings/utils'

describe('Skills', () => {
  let wrapper

  beforeEach(() => {
    window.gon = { skills: create('skills') }

    wrapper = mount(Skills)
  })

  it('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('has data properties', () => {
    const expected = ['skills', 'I18n']
    const received = Object.keys(wrapper.vm.$data)
    expect(received).toEqual(expected)
  })

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('Child components', () => {
    it('includes Skill', () => {
      const child_component = wrapper.find(Skill)
      expect(child_component.isVueInstance()).toBeTruthy()
    })

    it('renders 2 Skill objects', () => {
      wrapper = shallowMount(Skills)
      const skills = wrapper.findAll(Skill)
      expect(skills.length).toBe(2)
    })
  })
})
