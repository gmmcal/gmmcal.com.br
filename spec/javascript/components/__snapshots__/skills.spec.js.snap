// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills renders correctly 1`] = `
<section id="skills" class="technical">
  <div class="container">
    <div class="heading">
      <h2>Technical Skills</h2>
      <p>The subjects I'm currently under my radar, but I'm always open to new technologies.</p>
    </div>
    <div class="row">
      <p class="skill" style="font-size: 70px;">Skill Name</p>
      <p class="skill" style="font-size: 63px;">Another Skill</p>
    </div>
  </div>
</section>
`;
